src/App/mermaidData/pullApiData.js

```diff
@@ -44,80 +62,111 @@ export const pullApiData = async ({
-                dexiePerUserDataInstance,
-              })
-
-            const removeOfflineReadyProjectStatus =
-              dexiePerUserDataInstance.uiState_offlineReadyProjects.delete(projectId)
-
-            await Promise.all([
-              deleteProjectRelatedRecords,
-              resetProjectRelatedLastRevisionNumbers,
-              removeOfflineReadyProjectStatus,
-            ])
+  const _evaluatePullResponseAndUpdateOfflineMermaidDataAccordingly =
+    await dexiePerUserDataInstance.transaction(
+      'rw',
+      dexiePerUserDataInstance.benthic_attributes,
+      dexiePerUserDataInstance.choices,
+      dexiePerUserDataInstance.collect_records,
+      dexiePerUserDataInstance.fish_families,
+      dexiePerUserDataInstance.fish_genera,
+      dexiePerUserDataInstance.fish_species,
+      dexiePerUserDataInstance.project_managements,
+      dexiePerUserDataInstance.project_profiles,
+      dexiePerUserDataInstance.project_sites,
+      dexiePerUserDataInstance.projects,
+      dexiePerUserDataInstance.uiState_lastRevisionNumbersPulled,
+      dexiePerUserDataInstance.uiState_offlineReadyProjects,
+      async () => {
+        await persistLastRevisionNumbersPulled({
+          dexiePerUserDataInstance,
+          apiData,
+          projectId,
+        })
+
+        apiDataNamesToPull.forEach(async (apiDataType) => {
+          if (apiDataType === 'choices') {
+            // choices deletes property will always be empty, so we just ignore it
+            // additionally the updates property is an object, not an array, so we just store it directly
+
+            dexiePerUserDataInstance.choices.put({
+              id: 'enforceOnlyOneRecordEverStoredAndOverwritten',
+              choices: { ...apiData.choices?.updates, uiState_pushToApi: false },
+            })
+          }
+          if (apiDataType !== 'choices') {
+            const updates = apiData[apiDataType]?.updates ?? []
+            const updatesWithPushToApiTagReset = resetPushToApiTagFromItems(updates)
+            const deletes = apiData[apiDataType]?.deletes ?? []
+            const removes = apiData[apiDataType]?.removes ?? []
+            const deleteIds = deletes.map(({ id }) => id)
+            const removeIds = removes.map(({ id }) => id)
+            const error = apiData[apiDataType]?.error
+            const is401Or403 = error && [401, 403].includes(error.code)
+            const is403 = error?.code === 403
+
+            if (is403) {
+              shouldHandleUserDeniedSyncPull = true
+            }
+
+            const isDataTypeProjectAssociated = getIsDataTypeProjectAssociated(apiDataType)
+
+            const bulkDeleteIdsWithNoDuplicates = Array.from(new Set([...deleteIds, ...removeIds]))
+
+            dexiePerUserDataInstance[apiDataType].bulkPut(updatesWithPushToApiTagReset)
+            dexiePerUserDataInstance[apiDataType].bulkDelete(bulkDeleteIdsWithNoDuplicates)
+
+            if (is401Or403 && isDataTypeProjectAssociated && projectId) {
```

Comment:
in this conditional, I removed deleting the project from the offline-ready projects list, instead I set the `shouldDeleteProjectFromBeingConsideredOfflineReady = true`. The actual deleting needs to happen _after_ the app has redirected to a non-project page (to avoid the project being readded to the offline-ready projects list)

Reply:
makes sense!

Reply:
another naming option ... `shouldProjectBeRemovedFromOfflineReadyStatus` ?

would be good to throw in the comment about prepping to delete after the app has redirected in the code